### DISCUSSION TITLE
Issues running the tutorial file

### DISCUSSION THREAD

**Discussion by raymondchien** [mdolab/dafoam#423] [General]
Hi, I'm having some issues running the tutorial file. 
**OS: Ubuntu 20.04.6 LTS**
**Docker version: 23.0.4, build f480fb1**

Running the following command and getting the issue : 
```
dafoamuser@3a6d2b48104c:~/mount$ sudo ./preProcessing.sh
OpenFOAM environment not found, forgot to source the OpenFOAM bashrc?
```
According to this it can't find OpenFOAM? However when I check if it can find OpenFOAM I get:
```
dafoamuser@3a6d2b48104c:~/mount$ [ -z "$WM_PROJECT" ] && echo "WM_PROJECT is empty" || echo "WM_PROJECT is not empty"

WM_PROJECT is not empty
```

**Comment by friedenhe** [comment]
Can you try to run preProcessing.sh without the sudo command?

**Reply by raymondchien** [reply]
I get the following error:

> dafoamuser@d59fee1a6410:~/mount$ ./preProcessing.sh
/bin/bash: /home/dafoamuser/dafoam/packages/miniconda3/lib/libtinfo.so.6: no version information available (required by /bin/bash)
Generating mesh..
./preProcessing.sh: line 11: logMeshGeneration.txt: Permission denied
./preProcessing.sh: line 12: logMeshGeneration.txt: Permission denied
./preProcessing.sh: line 13: logMeshGeneration.txt: Permission denied
./preProcessing.sh: line 14: logMeshGeneration.txt: Permission denied
./preProcessing.sh: line 15: logMeshGeneration.txt: Permission denied
Generating mesh.. Done!
cp: cannot create directory '0': Permission denied


which is interesting because it seems like it sees the openfoam environment when sudo is not run?

**Comment by friedenhe** [comment]
Check the DAFoam website FAQ. It has a solution to permission-denied problems.

**Reply by raymondchien** [reply]
Hmm, I gave permissions to the entire tmpRun folder while in sudo, but I got additional errors. I fixed this by using chmod to give permissions to the additional directories mentioned in the error message. Everything works now. 

`dafoamuser@485bb0995998:~/mount/tmpRun$ ./preProcessing.sh
/bin/bash: /home/dafoamuser/dafoam/packages/miniconda3/lib/libtinfo.so.6: no version information available (required by /bin/bash)
Generating mesh..


--> FOAM FATAL ERROR: 
The parent directory does not allow write permission to the process,
or one of the directories in "/home/dafoamuser/mount/tmpRun/constant/polyMesh" did not allow search (execute) permission.

    From function bool Foam::mkDir(const Foam::fileName&, mode_t)
    in file POSIX.C at line 474.

FOAM exiting



--> FOAM FATAL ERROR: 
Cannot find file "points" in directory "polyMesh" in times "0" down to constant

    From function virtual Foam::IOobject Foam::fileOperation::findInstance(const Foam::IOobject&, Foam::scalar, const Foam::word&) const
    in file global/fileOperations/fileOperation/fileOperation.C at line 879.

FOAM exiting



--> FOAM FATAL ERROR: 
Cannot find file "points" in directory "polyMesh" in times "0" down to constant

    From function virtual Foam::IOobject Foam::fileOperation::findInstance(const Foam::IOobject&, Foam::scalar, const Foam::word&) const
    in file global/fileOperations/fileOperation/fileOperation.C at line 879.

FOAM exiting



--> FOAM FATAL ERROR: 
Cannot find file "points" in directory "polyMesh" in times "0" down to constant

    From function virtual Foam::IOobject Foam::fileOperation::findInstance(const Foam::IOobject&, Foam::scalar, const Foam::word&) const
    in file global/fileOperations/fileOperation/fileOperation.C at line 879.

FOAM exiting

Generating mesh.. Done!
`
